In [1]:
import csv

In [2]:
with open("trees.csv", newline='') as f:
    df = csv.reader(f)
    data = list(df)

1.Pour commencer, identifiez tous les genres botaniques représentés à Grenoble.
Combien y en a-t-il ? Essayez plusieurs manières de résoudre ce problème.

In [3]:
data[0]

['ELEM_POINT_ID',
 'CODE',
 'NOM',
 'GENRE',
 'GENRE_DESC',
 'CATEGORIE',
 'CATEGORIE_DESC',
 'SOUS_CATEGORIE',
 'SOUS_CATEGORIE_DESC',
 'CODE_PARENT',
 'CODE_PARENT_DESC',
 'ADR_SECTEUR',
 'BIEN_REFERENCE',
 'GENRE_BOTA',
 'ESPECE',
 'VARIETE',
 'STADEDEDEVELOPPEMENT',
 'EQUIPE',
 'REMARQUES',
 'ANNEEDEPLANTATION',
 'RAISONDEPLANTATION',
 'TRAITEMENTCHENILLES',
 'COURRIER',
 'IDENTIFIANTPLU',
 'TYPEIMPLANTATIONPLU',
 'INTITULEPROTECTIONPLU',
 'ANNEEABATTAGE',
 'ESSOUCHEMENT',
 'DIAMETREARBRE',
 'CAUSEABATTAGE',
 'COLLECTIVITE',
 'GeoJSON']

In [4]:
data[0].index('GENRE_BOTA')

13

In [5]:
genre_botanique = []
for i in range(len(data)):
    genre_botanique.append(data[i][13])
#genre_botanique

In [6]:
genre_botanique_unique = set(genre_botanique)
genre_botanique_unique.remove('GENRE_BOTA')
genre_botanique_unique.remove('')
genre_botanique_unique = list(genre_botanique_unique)
#genre_botanique_unique

In [7]:
a = 0
for i in genre_botanique_unique:
    a += 1
a

112

Il y a 112 genre botanique differents a Grenoble

1.2

In [8]:
genre_botanique_unique2 = []
for i in genre_botanique:
    if i not in genre_botanique_unique2:
        genre_botanique_unique2.append(i)
#genre_botanique_unique2

genre_botanique_unique2.pop(0)

In [9]:
a = 0
for i in genre_botanique_unique2:
    a += 1
a

114

2.Identifiez maintenant toutes les espèces différentes présentes à Grenoble.
N’oubliez pas que les espèces sont une sous catégorie de genre botanique !
(l’espèce est ‘Platanus Acerifolia’ et pas uniquement ‘Acerifolia’)

In [10]:
data[0]

['ELEM_POINT_ID',
 'CODE',
 'NOM',
 'GENRE',
 'GENRE_DESC',
 'CATEGORIE',
 'CATEGORIE_DESC',
 'SOUS_CATEGORIE',
 'SOUS_CATEGORIE_DESC',
 'CODE_PARENT',
 'CODE_PARENT_DESC',
 'ADR_SECTEUR',
 'BIEN_REFERENCE',
 'GENRE_BOTA',
 'ESPECE',
 'VARIETE',
 'STADEDEDEVELOPPEMENT',
 'EQUIPE',
 'REMARQUES',
 'ANNEEDEPLANTATION',
 'RAISONDEPLANTATION',
 'TRAITEMENTCHENILLES',
 'COURRIER',
 'IDENTIFIANTPLU',
 'TYPEIMPLANTATIONPLU',
 'INTITULEPROTECTIONPLU',
 'ANNEEABATTAGE',
 'ESSOUCHEMENT',
 'DIAMETREARBRE',
 'CAUSEABATTAGE',
 'COLLECTIVITE',
 'GeoJSON']

In [11]:
data[0].index('ESPECE')

14

In [12]:
espece = []
for i in range(len(data)):
    espece.append(data[i][14])
#espece

In [69]:
espece_unique = []
for i in espece:
    if i not in espece_unique:
        espece_unique.append(i)
#espece_unique

espece_unique.pop(0)

3.En utilisant la structure de données Dict compter le nombre de représentants de
chaque genre botanique

In [14]:
dict_genre_botanique = {}
for i in genre_botanique:
    if i not in dict_genre_botanique:
        dict_genre_botanique[i] = 1
    else:
        dict_genre_botanique[i] += 1
#dict_genre_botanique

del dict_genre_botanique['GENRE_BOTA']
del dict_genre_botanique['']

3.1Quel est le genre botanique le plus représenté ?

In [15]:
dict_genre_botanique = dict(sorted(dict_genre_botanique.items(), key=lambda x: x[1], reverse=True))
#dict_genre_botanique

Le genre botanique le plus représenté est le Acer avec 5276 arbre

3.2Listez tous les représentants uniques de leurs genre botanique

In [16]:
arbre_genre_unique = []
for i in dict_genre_botanique:
    if dict_genre_botanique[i] == 1:
        arbre_genre_unique.append(i)
#arbre_genre_unique

3.3Classer tous les genres botaniques en fonction de leur nombre (ordre
décroissant)

In [17]:
dict_genre_botanique = dict(sorted(dict_genre_botanique.items(), key=lambda x: x[1], reverse=True))
#dict_genre_botanique

3.4Quelle est la proportion du genre botanique le plus représenté parmi tous les
genres ?

In [18]:
nombre_total_genre = 0
for i in dict_genre_botanique.values():
    nombre_total_genre += i

In [19]:
5276 / nombre_total_genre

0.16886442196901805

La proportion du genre botanique le plus représenté est de 17%

3.5Quelle est la proportion de chacun des genres botaniques par rapport au
total ?

In [20]:
pourcent_genre_botanique = dict_genre_botanique.copy()
for i in pourcent_genre_botanique:
    pourcent_genre_botanique[i] /= nombre_total_genre
#pourcent_genre_botanique

In [21]:
sum(pourcent_genre_botanique.values())

1.0

3.6Les objectifs de la Métro sont-ils respectés en termes de diversité ?

3.7Idem précédemment mais en comptant les espèces.

In [22]:
dict_espece = {}
for i in espece:
    if i not in dict_espece:
        dict_espece[i] = 1
    else:
        dict_espece[i] += 1
#dict_espece

del dict_espece['ESPECE']
del dict_espece['']

In [23]:
nombre_total_espece = 0
for i in dict_espece.values():
    nombre_total_espece += i

In [24]:
pourcent_espece = dict_espece.copy()
for i in pourcent_espece:
    pourcent_espece[i] /= nombre_total_espece
#pourcent_espece

In [25]:
sum(pourcent_espece.values())

1.0

In [26]:
pourcent_espece = dict(sorted(pourcent_espece.items(), key=lambda x: x[1], reverse=True))
#pourcent_espece

Les objectifs de la Métro sont respectés en termes de diversité car il n'y a pas plus de 20% d'arbre du même genre botanique.
Cependant pour ce qui est des especes certains arbres de la même espece sont representée a plus de 10%. Ce qui ne respecte pas les objectifs de la Métro.

4.(plus dur) Comment la pratique de plantation a-t-elle évolué au cours du temps ?
Diversifie-t-on plus maintenant qu’avant ?

In [74]:
len(espece_unique)

255

In [75]:
len(genre_botanique_unique)

112

In [29]:
data[0]

['ELEM_POINT_ID',
 'CODE',
 'NOM',
 'GENRE',
 'GENRE_DESC',
 'CATEGORIE',
 'CATEGORIE_DESC',
 'SOUS_CATEGORIE',
 'SOUS_CATEGORIE_DESC',
 'CODE_PARENT',
 'CODE_PARENT_DESC',
 'ADR_SECTEUR',
 'BIEN_REFERENCE',
 'GENRE_BOTA',
 'ESPECE',
 'VARIETE',
 'STADEDEDEVELOPPEMENT',
 'EQUIPE',
 'REMARQUES',
 'ANNEEDEPLANTATION',
 'RAISONDEPLANTATION',
 'TRAITEMENTCHENILLES',
 'COURRIER',
 'IDENTIFIANTPLU',
 'TYPEIMPLANTATIONPLU',
 'INTITULEPROTECTIONPLU',
 'ANNEEABATTAGE',
 'ESSOUCHEMENT',
 'DIAMETREARBRE',
 'CAUSEABATTAGE',
 'COLLECTIVITE',
 'GeoJSON']

In [30]:
data[0].index('ANNEEDEPLANTATION')

19

In [31]:
data[0].index('ESPECE')

14

In [32]:
annee_espece = []
for i in range(len(data)):
    annee_espece.append((data[i][19], data[i][14]))
#annee_espece

In [49]:
annee_espece = sorted(annee_espece, key=lambda x: x[0])
#annee_espece

In [57]:
for i in annee_espece:
    if i == ('', ''):
        annee_espece.remove(i)
#annee_espece

In [76]:
dict_annee_espece = {}
liste_espece = []
for i in range(len(annee_espece)):
    a = annee_espece[i][0]
    e = annee_espece[i][1]
    if a not in dict_annee_espece:
        dict_annee_espece[a] = 1
        liste_espece = []
    elif e not in liste_espece:
        dict_annee_espece[a] += 1
        liste_espece.append(e)
dict_annee_espece

{'1950': 8,
 '1955': 7,
 '1960': 74,
 '1961': 2,
 '1964': 2,
 '1965': 56,
 '1966': 25,
 '1967': 20,
 '1968': 21,
 '1969': 37,
 '1970': 71,
 '1971': 22,
 '1972': 31,
 '1973': 10,
 '1974': 15,
 '1975': 87,
 '1976': 51,
 '1977': 44,
 '1978': 40,
 '1979': 34,
 '1980': 68,
 '1981': 49,
 '1982': 50,
 '1983': 50,
 '1984': 37,
 '1985': 54,
 '1986': 30,
 '1987': 42,
 '1988': 22,
 '1989': 35,
 '1990': 33,
 '1991': 26,
 '1992': 33,
 '1993': 36,
 '1994': 39,
 '1995': 49,
 '1996': 49,
 '1997': 21,
 '1998': 22,
 '1999': 56,
 '2000': 43,
 '2001': 47,
 '2002': 32,
 '2003': 47,
 '2004': 49,
 '2005': 41,
 '2006': 52,
 '2007': 29,
 '2008': 38,
 '2009': 76,
 '2010': 56,
 '2011': 91,
 '2012': 64,
 '2013': 61,
 '2014': 106,
 '2015': 112,
 '2016': 64,
 '2017': 44}

In [85]:
sum(dict_annee_espece.values()) / len(dict_annee_espece)

43.275862068965516

On peut observer que la moyenne de plantations d'especes différentes par année est d'environ 43. Avant 1975 cette moyenne n'etait que très peu respecter. Un pic a eu lieu entre 1975 et 1985. Cependant depuis les 9 dernières année de ce dataset nous pouvons observer un nette hausse au dessus de la moyenne. Avec un pic a 106 et 112 espece différentes en 2014 et 2015. Ce qui signifie que la diversité est plus abondante aujourd'hui, qu'il y a 10 ans ou 70 ans. Malgré ça cette diversité de plantation cette varie énormement au cours du temps.